src/module-elasticsuite-core/Model/System/Message/WarningAboutClusterShardsMisconfig.php

```diff
@@ -0,0 +1,169 @@
+     * @param UrlInterface        $urlBuilder         Url builder
+     */
+    public function __construct(
+        IndexSettingsHelper    $indexSettingHelper,
+        IndexStatsProvider $indexStatsProvider,
+        UrlInterface $urlBuilder
+    ) {
+        $this->helper             = $indexSettingHelper;
+        $this->indexStatsProvider = $indexStatsProvider;
+        $this->urlBuilder         = $urlBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isDisplayed()
+    {
+        $numberOfShards = $this->helper->getNumberOfShards();
+        $indexMaxSize = $this->getElasticsuiteIndexMaxSize()['size_in_bytes'];
+
+        if ($numberOfShards > 1) {
+            /* 10 Gb => 10737418240 bytes (in binary) */
+            if ($indexMaxSize < '10737418240') {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentity()
+    {
+        return hash('sha256', 'ELASTICSUITE_SHARDS_WARNING');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSeverity()
+    {
+        return self::SEVERITY_MAJOR;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getText()
+    {
+        $messageDetails = '';
+
+        // @codingStandardsIgnoreStart
+        $messageDetails .= __('The number of shards configured for Elasticsuite is incorrect.') . ' ';
+        $messageDetails .= __(
+            'You do not need to use <strong>%1 shards</strong> since your biggest Elasticsuite index is only <strong>%2</strong>.',
+            $this->helper->getNumberOfShards(),
+            $this->getElasticsuiteIndexMaxSize()['human_size']
+        ) . '<br/>';
+        $messageDetails .= __(
+            'Click here to go to the <a href="%1"><strong>Elasticsuite Config</strong></a> page and change your <strong>Number of Shards per Index</strong> parameter according to our <a href="%2" target="_blank"><strong>wiki page</strong></a>.',
+            $this->getElasticsuiteConfigUrl(),
+            self::ES_INDICES_SETTINGS_WIKI_PAGE
+        );
+        // @codingStandardsIgnoreEnd
+
+        return $messageDetails;
+    }
+
+    /**
+     * Get size of the biggest Elasticsuite Indices
+     *
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getElasticsuiteIndexMaxSize()
+    {
+        $elasticsuiteIndices = $this->indexStatsProvider->getElasticSuiteIndices();
+        $indexSizes = [];
+
+        foreach ($elasticsuiteIndices as $indexName => $indexAlias) {
+            $indexData = $this->indexStatsProvider->indexStats($indexName, $indexAlias);
+            $indexSizes[] = ['human_size' => $indexData['size'], 'size_in_bytes' => $indexData['size_in_bytes']];
+        }
+
+        $indexSizesInBytes = array_column($indexSizes, "size_in_bytes");
+        array_multisort($indexSizesInBytes, SORT_DESC, $indexSizes);
+
+        return current($indexSizes);
```

Comment:
How does it work if $indexSizes is empty ?

Reply:
Fixed.